account-analytics/src/main/java/com.bobocode/AccountAnalytics.java

```diff
@@ -113,7 +134,10 @@ public BigDecimal calculateTotalBalance() {
      * @return a map where key is a creation month and value is total balance of all accounts created in that month
      */
     public Map<Month, BigDecimal> groupTotalBalanceByCreationMonth() {
-        throw new UnsupportedOperationException("It's your job to implement this method"); // todo
+        return accounts.stream()
+                .collect(groupingBy(a -> a.getCreationDate().getMonth(),
+                        mapping(Account::getBalance,
+                                reducing(BigDecimal.ZERO, BigDecimal::add))));
```

Comment:
The value of `identity` will be used as a first value, when it reduces your stream. In case your stream is empty, `reduce()` will return `identity` value.